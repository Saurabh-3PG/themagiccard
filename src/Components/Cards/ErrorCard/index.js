import React from "react";
import { Box, Typography } from "@material-ui/core";
import {ERROR_MSG} from "../../../utility/constants";
import PropTypes from "prop-types";
class ErrorCard extends React.Component {
  render() {
    const { componentID } = this.props;
    const uniqueId = componentID + "_ErrorCard";
    return (
      <Box className="ErrorCard" id={uniqueId} style={{maxWidth: "100%"}}>
          <Typography className="font fontBold fontSize_xl m_before_xxxl m_after_xxxl" component="h2" style={{
                    minHeight: "150px",
                    width: "640px",
                    maxWidth: "100%",
                    display: "flex",
                    alignItems: "center",
                    justifyContent: "center",
                    color: "red",
                    boxShadow: "2px 4px 12px -4px rgb(0, 0, 0, 0.42)",
                    padding: "64px",
                    textAlign: "center",
          }}>
            {ERROR_MSG}
          </Typography>
      </Box>
    );
  }
}

ErrorCard.propTypes = {
  componentID: PropTypes.string.isRequired
};

export default ErrorCard;
