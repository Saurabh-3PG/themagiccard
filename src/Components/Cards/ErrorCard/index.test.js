import React from 'react';
import renderer from 'react-test-renderer';
import ErrorCard from './index';

test('ErrorCard app component', () => {
  const component = renderer.create(
    <ErrorCard
      componentID={'ErrorCard'}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});