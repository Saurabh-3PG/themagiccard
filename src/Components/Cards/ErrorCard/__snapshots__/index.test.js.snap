// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCard app component 1`] = `
<div
  className="MuiBox-root MuiBox-root-1 ErrorCard"
  id="ErrorCard_ErrorCard"
  style={
    Object {
      "maxWidth": "100%",
    }
  }
>
  <h2
    className="MuiTypography-root font fontBold fontSize_xl m_before_xxxl m_after_xxxl MuiTypography-body1"
    style={
      Object {
        "alignItems": "center",
        "boxShadow": "2px 4px 12px -4px rgb(0, 0, 0, 0.42)",
        "color": "red",
        "display": "flex",
        "justifyContent": "center",
        "maxWidth": "100%",
        "minHeight": "150px",
        "padding": "64px",
        "textAlign": "center",
        "width": "640px",
      }
    }
  >
    No is no result found for the selected filter, Please change filter or use clear filter reaset it.
  </h2>
</div>
`;
